### DISCUSSION TITLE
Properly connect to remote server

### DISCUSSION THREAD

**Discussion by victor-cali** [microsoft/vscode-jupyter#10839] [Questions and Answers]
Hello everyone.

I am trying to get up and running a Jupyter remote server I can acces from my local VS Code.  The server seems to be ready, I can access it from my browser and my Windows Terminal SSH. I based mostly on this [tutorial](https://gianniceresa.com/2019/12/jupyter-sandbox-in-an-oracle-cloud-compute-instance/) to get the server running, with a few changes due to the use of `jupyter server` instead of `jupyter lab`.

The tutorial got me through creating the server, however now I have problems accessing it with VS Code. The server uses password and a ssl certificate so it has no token. the host I input to vscode looks something like this: **https://server-ip.server-port**. Such address will take me to an empty screen with the legend: _A Jupyter Server is running._

If I write the endpoint **/lab?** then I will enter Jupyter Lab UI. I have tried both with and without such endpoint (it always cuts the endpoint out and adds **/?token=null**) and it does seem to connect at first, at least that is what it says at the bottom right corner with the option _Jupyter Server:_ marked as **remote**. However, in the kernel select menu at the upper right corner, I cannot see the environments of the remote server but only my local environments, when I run `!hostname` command, I can see I am still in my local machine.

PS. I should add I use anaconda both in my local and remote for environments, never had problems with the local, and the remote ones I can see them at the browser UI. 

Any advice? :(

**Comment by rchiodo** [comment]
What does the jupyter output show? 

If there's no token provided we try to query the server to see if it needs a user name (I believe by just asking for the [/tree](https://github.com/microsoft/vscode-jupyter/blob/c68a25624fb3f46656795ef8a4d66a2db8e368ca/src/kernels/jupyter/launcher/jupyterPasswordConnect.ts#L354) subfolder) but this may not work if your server doesn't respond the way we expect.

**Comment by amunger** [comment]
I think someone on the team just hit this as well and should be fixed in the next release: https://github.com/microsoft/vscode-jupyter/pull/10963

you could work around it by running `jupyter notebook` instead of `jupyter lab`, and using the server URL from that output
